dali/imgcodec/image_format_test.cc

```diff
@@ -75,6 +77,50 @@ class ImageFormatTest : public ::testing::Test {
   std::vector<char> data_;
 };
 
+class ComparisonTestBase : public ImageFormatTest {
+ protected:
+  virtual TensorShape<> ShapeOf(const std::string &filenames) const = 0;
+
+  void Run(const std::string &filename, std::string expected_format) {
+    Test(filename, expected_format, ShapeOf(filename));
+  }
+
+ public:
+  void RunOnDirectory(std::string directory, std::string expected_format,
+                      std::vector<std::string> extensions) {
+    unsigned nimages = 0;
+
+    for (const auto &entry : std::filesystem::recursive_directory_iterator(directory)) {
+      if (entry.is_regular_file()) {
+        const auto path = entry.path().string();
+        for (const auto& ext : extensions) {
+          if (path.substr(path.size() - ext.size(), ext.size()) == ext) {
+            Run(path, expected_format);
+            nimages++;
+          }
+        }
+      }
+    }
+
+    if (nimages == 0)
+      FAIL() << "No matching images in " << directory;
+  }
+};
+
+class CompatibilityTest : public ComparisonTestBase {
```

Comment:
Can you please add doxygen saying what this tests does (comparing against the previous implementation)?

Reply:
Done